### PR TITLE
Adding CosmosDB and Functions App deployment. 

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
I develop three files for deployment. 

* Build script for Azure Functions Leaderboard backend
* Testing deployment of CosmosDB (it works)
* Deploy FunctionApp with configuration with deploying app which we build at the first dot

Since we need `zip` command for packaging it, I add zip command to the proctorVM deployment script. 

This is the progress report of mine.  One of the problem of for the endpoint URL is CORS. CORS settings doens't support via Auzre CLI. we might use ARM template instead. [Azure Functions: how to set CORS via automation?
](https://stackoverflow.com/questions/46921887/azure-functions-how-to-set-cors-via-automation) 

* [Azure cli functionapp and webapp has no support for CORS control #4565](https://github.com/Azure/azure-cli/issues/4565)

However, this strategy works except for the CORS part. For the CORS setting, just swtich function deployment / config part to the arm template. 

```
Break deploy.sh into 2 scripts (Tsuyoshi)
    - install cosmosdb

    - build function and zip  -> Done.
    - install function onto azure -> Done.
    - configure function to talk to cosmos db -> Done.
    - output function endpoint URI for sentinel -> NotDone.

    - HELM chart for sentinel installation DONE

api (Tsuyoshi)
    - build from source -> Done
```

### CODE REVIEW

File: provision-proctor/setup.sh [hunk 1]
```diff
@@ -174,5 +191,11 @@ bash ./build_deploy_sentinel.sh -r $resourceGroupProctor -g $registryName -n ${t
 echo "6-Build and deploy leaderboard website to AKS  (bash ./build_deploy_web.sh -m $proctorName -d <dnsURL>)"
 bash ./build_deploy_web.sh -m $proctorName -d $dnsURL
 
-echo "7-Clean the working environment"
+echo "7-Provision CosmosDB"
```
Comment (dtzar) [inline provision-proctor/setup.sh]: We should have this and Azure Functions before so we can pass the value into Sentinel and Leaderboard.

File: provision-proctor/deploy_function.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+cosmosdbEndpoint=$(az cosmosdb show --name $cosmosDBName  --resource-group $resourceGroupName --query documentEndpoint --output tsv)
+cosmosdbKey=$(az cosmosdb list-keys --name $cosmosDBName  --resource-group $resourceGroupName --query primaryMasterKey --output tsv)
+
+# create a new function app, assign it to the resource group you have just created
+az functionapp create --name $functionAppName --resource-group $resourceGroupName --storage-account $storageAccountName --consumption-plan-location $location
+
+
+# configure function app settings to use cosmosdb connection string
+# This part include Run-From-Zip deployment
+# TODO:
```
Comment (dtzar) [inline provision-proctor/deploy_function.sh]: Cleanup TODO and add backlog task.

File: provision-proctor/deploy_function.sh [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+echo $zipFileName
+echo -e "\n"
+
+
+# create a storage account 
+az storage account create --name $storageAccountName --location $location  --resource-group $resourceGroupName --sku Standard_LRS
+
+# upload
+
+# TODO upload zipfile to the storage account.
```
Comment (dtzar) [inline provision-proctor/deploy_function.sh]: Clean up comments

**Comment by rguthriemsft** [comment]
I put my approval in pending david's suggestions.  Let me know once merged.

**Pull request merged**